stablehlo/reference/Ops.cpp

```diff
@@ -372,6 +403,56 @@ Tensor evalSliceOp(const Tensor &operand, Index startIndices, Sizes strides,
   return result;
 }
 
+SmallVector<Tensor> evalSortOp(ArrayRef<Tensor> inputs, Axis dimension,
+                               bool isStable, Region &comparator,
+                               Scope &scope) {
+  SmallVector<Tensor> runtimeResults;
+  for (auto input : inputs) runtimeResults.push_back(Tensor(input.getType()));
+  if (dimension < 0) dimension += inputs[0].getRank();
+  SmallVector<int64_t> indices(inputs.front().getShape()[dimension]);
+  auto inputShape = inputs.front().getShape();
+  inputShape[dimension] = 1;
+  auto inputSizes = Sizes(inputShape);
+  for (auto resultIt = IndexSpaceIterator(inputSizes, Index(inputShape.size()));
+       resultIt != IndexSpaceIterator(inputSizes, std::nullopt); ++resultIt) {
+    std::iota(indices.begin(), indices.end(), 0);
+    auto compare = [&](int64_t lhs, int64_t rhs) {
+      SmallVector<Tensor> args;
+      auto lhsIndex = *resultIt;
+      auto rhsIndex = *resultIt;
+      lhsIndex[dimension] = lhs;
+      rhsIndex[dimension] = rhs;
+      for (auto input : inputs) {
+        auto type =
+            comparator.front().getArgumentTypes()[0].dyn_cast<TensorType>();
+        auto lhs = Tensor(type);
+        auto rhs = Tensor(type);
+        lhs.set({}, input.get(lhsIndex));
+        rhs.set({}, input.get(rhsIndex));
+        args.push_back(lhs);
+        args.push_back(rhs);
+      }
+      auto cmpResult = eval(comparator, args, &scope);
+      return cmpResult[0].get({}).getBooleanValue();
+    };
+
+    if (isStable)
+      std::stable_sort(indices.begin(), indices.end(), compare);
+    else
+      std::sort(indices.begin(), indices.end(), compare);
+    auto resultIdx = *resultIt;
```

Comment:
We can be simplified and improve readability if we keep separate copy for resultIdx and operandIdx. 
PTAL suggestion below